WordPress/openverse
# Use contentlist role for search results for a11y

## Fixes



Fixes #794 by @zackkrida
Fixes #541 by @obulat 

## Description


This PR wraps the search results into an ordered list to make the screen reader announce it as a contentlist, and to make it possible for the users to know how many items there are.

For the list `aria-label`, I used "Search results for query", without the media type. I think this might be clear from previous navigation, and it's easier to implement this way. I am open to suggestions about the text.

There are also a couple of other fixes:
- image grid adds a Skeleton loader. This is a very simple addition that makes all of the search views consistent.
- "Openverse" text is passed to the home link to prevent it saying "Opening braces openverse closing braces"
- Removes the incorrect grid for the VContentLinks that I added in the grid fix PR (I accidentally added the grid fixes to both the "All content" results AND the content links).


## Testing Instructions

Run the app using `just frontend/run dev`
Open a search page `localhost:8443/search/image?q=cat` and turn on a screenreader (CMD + F5 for VoiceOver on Mac, for Windows I used the UI to open NVDA)
On Mac, with Voiceover:
When you navigate to the search results, you should hear "Content list Search results for cat, 20 items" when screenreader reads out the page, and "Entering Search results for cat content list" when you navigate to the results.

On Windows, with NVDA:
You should hear "Search results for cat List with 27 items Cat link" when you navigate to the first result on All content list.

## Checklist



[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
